client/driver/docker_progress_test.go

```diff
@@ -0,0 +1,52 @@
+package driver
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_DockerImageProgressManager(t *testing.T) {
+
+	pm := &imageProgressManager{
+		imageProgress: &imageProgress{
+			timestamp: time.Now(),
+			layers:    make(map[string]*layerProgress),
+		},
+	}
+
+	_, err := pm.Write([]byte(`{"status":"Pulling from library/golang","id":"1.9.5"}
+{"status":"Pulling fs layer","progressDetail":{},"id":"c73ab1c6897b"}
+{"status":"Pulling fs layer","progressDetail":{},"id":"1ab373b3deae"}
+`))
+	assert.NoError(t, err)
```

Comment:
We generally use `testify/require` instead of `testify/assert` as assert doesn't fail immediately. It's no big deal but in cases like this where you're checking for errors, the very next line is likely to panic and make for confusing test output if there is an error.